jwpincus/punstartr
# Add Two Endpoints for Internal API

#### What does  this PR do?

Adds API endpoints for projects with most backers and projects ending soonest.
Updates seed file (there were minor issues with seeding projects with city and country additions from last sprint).
Updates configuration of database_cleaner and factory girl in rails_helper.
Updates the README to include API documentation.

#### Where should the reviewer start?

Review api/v1/project controllers and then the project model methods that use active record to return property objects. I'd also review the project serializer to ensure the attributes I decided to return make sense.

#### How should this be manually tested?

Pull down branch.
Fire up local server. 
Ensure development database is seeded.
Use postman or chrome to test the following endpoints:

`GET /api/v1/projects/most_backers`
`GET /api/v1/projects/most_backers?limit=6`
`GET /api/v1/projects/ending_soon`
`GET /api/v1/projects/ending_soon?limit=6`

#### Any background context you want to provide?

Nah.

#### What are the relevant tickets?

https://www.pivotaltracker.com/story/show/146054575
https://www.pivotaltracker.com/story/show/146054685

#### Screenshots (if appropriate)
#### Questions:
  - Do Migrations Need to be ran? 
  - Do Environment Variables need to be set? 
  - Any other deploy steps? 

I would recommend `rake db:reset` so that the database is properly seeded.
